core/include/jiminy/core/robot/FixedFrameConstraint.h

```diff
@@ -0,0 +1,90 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+///
+/// \brief      Class representing a fixed frame constraint.
```

Comment:
Currently this is the only possible constraint: a fully fixed body. 
This may be generalized by allowing constraint of some axes only (e.g. to simulate an object sliding on a plane).

The other type of constraint I'm interested in is a wheel turning on the ground. At first I though I would do that outside of Jiminy: implement a ```ConstraintFunctor``` class that applies a constraint from a user-specified callback for the jacobian + drift. This allows an arbitrary constraint design from the user.

However, from our discussion I'm thinking ground / wheel is something rather common, so maybe it's a good thing to have an implementation inside Jiminy: this can be done in a generic manner I think, with the user specifying the wheel center, radius and  axis of rotation. Then maybe I don't even need the ```ConstraintFunctor``` class. What do you think ?

Reply:
I think having `ConstraintFunctor` is an interesting feature. 

As you said ground/wheel contact is rather common and perceived as "simple". I don't think and end-user expects having to define it manually. As you said, the input arguments could be "wheel center, radius, axis of rotation" but I think it may be interesting to also be able to set a "width" (equals to zero eventually) so that the wheel can stand on the ground with zero speed without falling.

Reply:
I'll implement this in a separate PR so we can merge this already large request.